shell/platform/darwin/ios/framework/Source/FlutterEngineGroup.mm

```diff
@@ -0,0 +1,60 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Headers/FlutterEngineGroup.h"
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterEngine_Internal.h"
+
+@interface FlutterEngineGroup ()
+@property(nonatomic, copy) NSString* name;
+@property(nonatomic, strong) NSMutableArray<NSValue*>* engines;
+@property(nonatomic, strong) FlutterDartProject* project;
+@end
+
+@implementation FlutterEngineGroup {
+  int _engineCount;
+}
+
+- (instancetype)initWithName:(NSString*)name project:(nullable FlutterDartProject*)project {
+  self = [super init];
+  if (self) {
+    self.name = name;
+    self.engines = [[NSMutableArray<NSValue*> alloc] init];
+    self.project = project;
+  }
+  return self;
+}
+
+- (void)dealloc {
+  NSNotificationCenter* center = [NSNotificationCenter defaultCenter];
+  [center removeObserver:self];
+  [_name release];
+  [_engines release];
+  [super dealloc];
+}
+
+- (FlutterEngine*)makeEngineWithEntrypoint:(nullable NSString*)entrypoint {
+  NSString* engineName = [NSString stringWithFormat:@"%@.%d", self.name, ++_engineCount];
+  FlutterEngine* engine;
+  if (self.engines.count <= 0) {
+    engine = [[FlutterEngine alloc] initWithName:engineName project:self.project];
```

Comment:
add some low level FML logs when making the first engines again?

Reply:
I don't think we do that anywhere else in the engine codebase.  They are against printing anything to console unless it's an error.  There isn't another log.

Reply:
Not very important, I just meant DLOG or or FML_LOG below LOG_ERROR don't actually print to console unless you turn on verbose right? https://github.com/flutter/engine/blob/405b105fe8b7ca745c1546f3bc129ad6f92e1b57/shell/common/shell.cc#L197